utils/cfutils.go

```diff
@@ -0,0 +1,203 @@
+				Plan         string      `json:"plan"`
+				Name         string      `json:"name"`
+				Tags         []string    `json:"tags"`
+				InstanceGUID string      `json:"instance_guid"`
+				InstanceName string      `json:"instance_name"`
+				BindingGUID  string      `json:"binding_guid"`
+				BindingName  interface{} `json:"binding_name"`
+				Credentials  struct {
+				} `json:"credentials"`
+				SyslogDrainURL interface{} `json:"syslog_drain_url"`
+				VolumeMounts   []struct {
+					ContainerDir string `json:"container_dir"`
+					Mode         string `json:"mode"`
+					DeviceType   string `json:"device_type"`
+				} `json:"volume_mounts"`
+			} `json:"fs-storage"`
+		} `json:"VCAP_SERVICES"`
+	} `json:"system_env_json"`
+	ApplicationEnvJSON struct {
+		VcapApplication struct {
+			CfAPI  string `json:"cf_api"`
+			Limits struct {
+				Fds int `json:"fds"`
+			} `json:"limits"`
+			ApplicationName  string      `json:"application_name"`
+			ApplicationUris  []string    `json:"application_uris"`
+			Name             string      `json:"name"`
+			SpaceName        string      `json:"space_name"`
+			SpaceID          string      `json:"space_id"`
+			OrganizationID   string      `json:"organization_id"`
+			OrganizationName string      `json:"organization_name"`
+			Uris             []string    `json:"uris"`
+			Users            interface{} `json:"users"`
+			ApplicationID    string      `json:"application_id"`
+		} `json:"VCAP_APPLICATION"`
+	} `json:"application_env_json"`
+}
+
+func readAppEnv(app string) ([]byte, error) {
+	guid, err := exec.Command("cf", "app", app, "--guid").Output()
+	if err != nil {
+		return nil, err
+	}
+
+	env, err := exec.Command("cf", "curl", fmt.Sprintf("/v3/apps/%s/env", strings.Trim(string(guid[:]), "\n"))).Output()
+	if err != nil {
+		return nil, err
+	}
+	return env, nil
+
+}
+
+func checkUserPathAvailability(app string, path string) (bool, error) {
+	output, err := exec.Command("cf", "ssh", app, "-c", "[[ -d \""+path+"\" && -r \""+path+"\" && -w \""+path+"\" ]] && echo \"exists and read-writeable\"").Output()
+	if err != nil {
+		return false, err
+	}
+
+	if strings.Contains(string(output[:]), "exists and read-writeable") {
+		return true, nil
+	}
+
+	return false, nil
+}
+
+func (checker CfJavaPluginUtilImpl) CheckRequiredTools(app string) (bool, error) {
+	output, err := exec.Command("cf", "ssh-enabled", app).Output()
+	if err != nil {
+		return false, err
+	}
+	if !strings.Contains(string(output[:]), "ssh support is enabled for app '"+app+"'.") {
+
+		return false, errors.New("ssh is not enabled for app: '" + app + "', please run below 2 shell commands to enable ssh and try again(please note application should be restarted before take effect):\ncf enable-ssh " + app + "\ncf restart " + app)
+	}
+
+	output, err = exec.Command("cf", "ssh", app, "-c", "find -executable -name jvmmon | head -1 | tr -d [:space:] || find -executable -name jmap | head -1 | tr -d [:space:]").Output()
```

Comment:
Unfortunately, this now works for the `sap_java_buildpack`, but not for the community Java buildpack anymore.
Although `jmap` exists:

```shell
$ find -executable -name jmap | head -1 | tr -d [:space:]
./app/.java-buildpack/open_jdk_jre/bin/jmap
```

it doesn't return any results if checked with the new command and hence exits:
```shell
$  find -executable -name jvmmon | head -1 | tr -d [:space:] || find -executable -name jmap | head -1 | tr -d [:space:]
[no result]
```

The problem is that `find` returns a zero exit code even if it doesn't find any file and therefore never executes the part after `||`.

Unfortunately, `find` doesn't support extended (`-E`) regex mode in the application container, so it's a little bit tricky to do it with plain `find`. What seems to work as a little workaround is using grep:

```shell
find -executable | grep -E '(.*jmap$)|(.*jvmmon$)'
```